processing-service-container/src/test/java/com/github/cafdataprocessing/processing/service/tests/GlobalConfigIT.java

```diff
@@ -0,0 +1,120 @@
+    
+    private static GlobalConfigApi globalConfigAPI;
+    
+    @BeforeClass
+    public static void setup() throws Exception {
+        final ApiClient apiClient = ApiClientProvider.getApiClient();
+        globalConfigAPI = new GlobalConfigApi(apiClient);
+    }
+    
+    @Test
+    public void createAndGetGlobalConfig() throws ApiException {
+        createConfig("1");
+        
+        final GlobalConfig expectedConfig = new GlobalConfig();
+        expectedConfig.setDefault(TEST_VALUE_1);
+        expectedConfig.setDescription(TEST_DESCRIPTION_1);
+        
+        final GlobalConfig actualConfig = globalConfigAPI.getGlobalConfig(TEST_KEY_1);
+        assertEquals(actualConfig, expectedConfig);
+    }
+    
+    @Test
+    public void getGlobalConfigOnNotExistingRecord() throws ApiException {
+        try {
+            globalConfigAPI.getGlobalConfig(TEST_NOT_EXISTING_KEY);
+        } catch (ApiException e) {
+            assertTrue(e.getCode() == 404);
+            assertTrue(e.getMessage().contains("Failed to find global config with key '" + TEST_NOT_EXISTING_KEY + "'."));
+        }
+    }
+    
+    @Test
+    public void getAllGlobalConfigsForExistigRecords() throws ApiException {
+        createConfig("1");
+        createConfig("2");
+                
+        assertEquals(2, globalConfigAPI.getGlobalConfigs().size());
+    }
+    
+    @Test(description = "Test that an empty array is returned and not a null.")
+    public void getAllGlobalConfigsForEmptyRecords() throws ApiException {
+        //delete all records from previous tests
+        globalConfigAPI.getGlobalConfigs().forEach(config -> {
+            try {
+                globalConfigAPI.deleteGlobalConfig(config.getKey());
+            } catch (ApiException ex) {
+                Logger.getLogger(GlobalConfigIT.class.getName()).log(Level.SEVERE, "Failed to delete all records.", ex);
+            }
+        });
+        
+        assertTrue(globalConfigAPI.getGlobalConfigs().isEmpty());
+    }
+    
+    @Test
+    public void deleteGlobalConfigOnExistingRecord() throws ApiException {
+        createConfig("1");
+        globalConfigAPI.deleteGlobalConfig(TEST_KEY_1);
+        
+        final GlobalConfigsEntry deletedConfig = new GlobalConfigsEntry();
+        deletedConfig.setKey(TEST_KEY_1);
+        deletedConfig.setDefault(TEST_VALUE_1);
+        deletedConfig.setDescription(TEST_DESCRIPTION_1);
+        
+        assertTrue(!globalConfigAPI.getGlobalConfigs().contains(deletedConfig));
+    }
+    
+    
+    @Test
+    public void deleteGlobalConfigOnNotExistingRecord() {
+        try {
+            globalConfigAPI.deleteGlobalConfig(TEST_NOT_EXISTING_KEY);
+        } catch (ApiException e) {
+            assertTrue(e.getCode() == 404);
+            assertTrue(e.getMessage().contains("Global config specified for delete using key '" + TEST_NOT_EXISTING_KEY + "' was not found."));
```

Comment:
This is OK for now but I would say avoid checking full error message text as that can be prone to change and would force an update of the text that the test needs to check. Really the 404 status is good enough here I think.